### PR TITLE
update to 3.6.2

### PR DESCRIPTION
**Pull request by Arishamays1**
# matplotlib version update 3.6.2
- updated sha256
- added a patch for qhull network issue which caused build failures for Linux-s390x 
- added scripting build.sh

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -19,45 +19,54 @@ outputs:
...
     build:
-      skip: true  # [py<37]
+      skip: true  # [py<38]
       ignore_run_exports:
         - tk  # [linux]
+        - zlib
     requirements:
       build:
+        - python
+        - cross-python_{{ target_platform }}       # [build_platform != target_platform]
```
Comment (marcoesters) [inline recipe/meta.yaml:29]: Not needed - we do not cross-compile

File: recipe/meta.yaml [hunk 2]
```diff
@@ -19,45 +19,54 @@ outputs:
...
+        - numpy
```
Comment (marcoesters) [inline recipe/meta.yaml:30]: Is numpy needed here when we have it in `host`? If so, why do we not have the same pinnings like we do in host?

File: recipe/meta.yaml [hunk 3]
```diff
@@ -19,45 +19,54 @@ outputs:
...
         # an explanation to the utilization of numpy can be found
         # here: https://github.com/AnacondaRecipes/matplotlib-feedstock/pull/13#issuecomment-1296704563
         - numpy   1.19  # [not (osx and arm64) and (py==37 or py==38 or py==39)]
         - numpy   1.21  # [(osx and arm64) and (py==38 or py==39)]
         - numpy   1.21  # [py==310]
         - setuptools
-        - setuptools_scm >=4
+        - setuptools_scm >=7
         - setuptools_scm_git_archive
         - tk  # [linux]
```
Comment (marcoesters) [inline recipe/meta.yaml:49]: Are they still needed? I do not see them in `setup.py`: https://github.com/matplotlib/matplotlib/blob/v3.6.2/setup.py#L303-L307

File: recipe/meta.yaml [hunk 4]
```diff
@@ -19,45 +19,54 @@ outputs:
...
+        - contourpy >=1.0.1
         - fonttools >=4.22.0
         - freetype >=2.3
         - kiwisolver >=1.0.1
+        - numpy >=1.19
         - {{ pin_compatible('numpy') }}
         - packaging >=20.0
         - pillow >=6.2.0
-        - pyparsing >=2.2.1
+        - pyparsing >=2.3.1
```
Comment (marcoesters) [inline recipe/meta.yaml:63]: `pyparsing` is still on version 2.2.1: https://github.com/matplotlib/matplotlib/blob/v3.6.2/setup.py#L316

File: recipe/meta.yaml [hunk 5]
```diff
@@ -76,15 +85,15 @@ outputs:
...
     requirements:
       host:
         - python
       run:
         - python
-        - pyqt  # [(linux64 or win) and python_impl == "cpython"]
-        - tornado
-        - {{ pin_subpackage('matplotlib-base', max_pin="x.x.x.x") }}
+        - pyqt  >=5  # [(linux64 or win) and python_impl == "cpython"]
+        - tornado >=5
```
Comment (marcoesters) [inline recipe/meta.yaml:95]: I can't find this requirement upstream. Can you provide a link as a comment so I can verify this?

File: recipe/meta.yaml (hunk 2 above)
Comment (Arishamays1) [reply recipe/meta.yaml:30]: Hey Marco! So the rationale for the pinnings is our artifacts for our different platforms run on different versions of numpy so it's best to use two different versions in the host section. [See here ](https://github.com/AnacondaRecipes/matplotlib-feedstock/pull/13#issuecomment-1296704563) for further explanation.

File: recipe/meta.yaml (hunk 1 above)
Comment (Arishamays1) [reply recipe/meta.yaml:29]: Got it, removed.

File: recipe/meta.yaml (hunk 2 above)
Comment (marcoesters) [reply recipe/meta.yaml:29]: This doesn't answer my questions. Let me rephrase:

* Why do we need numpy in build and in host at the same time?
* Why are the build pinnings different than in host?

File: recipe/meta.yaml (hunk 5 above)
Comment (cbouss) [reply recipe/meta.yaml:95]: The optional dependencies are listed here: https://matplotlib.org/stable/devel/dependencies.html

File: recipe/meta.yaml (hunk 2 above)
Comment (skupr-anaconda) [reply recipe/meta.yaml:30]: > python and numpy should be removed from host. 

From `build` **not** from `host`

Comment (cbouss) [reply recipe/meta.yaml:30]: @skupr-anaconda yes, thank you!

File: recipe/meta.yaml (hunk 5 above)
Comment (marcoesters) [reply recipe/meta.yaml:95]: Thank you!

File: recipe/meta.yaml (hunk 2 above)
Comment (Arishamays1) [reply recipe/meta.yaml]: Went ahead and removed `# [build_platform != target_platform]` , `python`, and `numpy` from the build section.

File: recipe/meta.yaml (hunk 4 above)
Comment (Arishamays1) [reply recipe/meta.yaml]: Changed the pinnings for `pyparsing` back to 2.2.1. Thanks!

File: recipe/meta.yaml [hunk 6]
```diff
@@ -19,45 +19,53 @@ outputs:
...
+        - wheel
       run:
         - python
+        - __osx >={{ MACOSX_DEPLOYMENT_TARGET|default("10.9") }}  # [osx and x86_64]
         - cycler >=0.10
+        - contourpy >=1.0.1
         - fonttools >=4.22.0
         - freetype >=2.3
         - kiwisolver >=1.0.1
+        - numpy >=1.19
```
[the remark this answers is not in the document]
Comment (Arishamays1) [reply recipe/meta.yaml]: Went ahead and made this amendment. Thanks Charles!

File: recipe/meta.yaml (hunk 3 above)
Comment (Arishamays1) [reply recipe/meta.yaml:46]: I've found some documentation regarding `tk` and it is an optional dependency for `matplotlib`. See [here](https://matplotlib.org/stable/devel/dependencies.html#:~:text=Tk%20(%3E%3D%208.4%2C%20!%3D%208.6.0%20or%208.6.1)%3A%20for%20the%20Tk%2Dbased%20backends.%20Tk%20is%20part%20of%20most%20standard%20Python%20installations%2C%20but%20it%27s%20not%20part%20of%20Python%20itself%20and%20thus%20may%20not%20be%20present%20in%20rare%20cases).
`Setuptools_scm_git_archive` seems to play an essential role in our successful builds. I've noticed that when removed from the host section, all our builds began to fail.

File: recipe/meta.yaml [hunk 7]
```diff
@@ -19,31 +19,37 @@ outputs:
...
+        - zlib
     requirements:
       build:
         - pkg-config  # [not win]
         - {{ compiler('c') }}
         - {{ compiler('cxx') }}
       host:
         - python
-        - pip
+        - pip >=9.0.1
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:33]: pip 9.0.1 is so old s you can remove pinning. cond-build always takes the latest version from host. You can make an experiment and check host logs.

File: recipe/meta.yaml [hunk 8]
```diff
@@ -52,12 +58,11 @@ outputs:
         - pillow >=6.2.0
         - pyparsing >=2.2.1
         - python-dateutil >=2.7
-
+        - freetype
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:61]: Missing empty line before `test` (for readability)

File: recipe/meta.yaml [hunk 9]
```diff
@@ -76,15 +81,15 @@ outputs:
...
+      skip: true  # [py<38]
     requirements:
       host:
         - python
       run:
         - python
-        - pyqt  # [(linux64 or win) and python_impl == "cpython"]
-        - tornado
-        - {{ pin_subpackage('matplotlib-base', max_pin="x.x.x.x") }}
+        - pyqt  >=5  # [(linux64 or win) and python_impl == "cpython"]
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:90]: pyqt`  `>=5 has a redundant space

File: recipe/meta.yaml (hunk 3 above)
Comment (skupr-anaconda) [reply recipe/meta.yaml:46]: @marcoesters Here is a great comment about `tk` as a default backend https://github.com/conda-forge/matplotlib-feedstock/pull/157#issue-325602391

File: recipe/meta.yaml (hunk 8 above)
Comment (marcoesters) [reply recipe/meta.yaml:61]: I would say adding an empty line would make it less readable. Since this is a multi-output recipe, I would only add blank lines between each output.

File: recipe/meta.yaml (hunk 9 above)
Comment (Arishamays1) [reply recipe/meta.yaml:90]: Thanks, Serhii! Removed it.

File: recipe/meta.yaml (hunk 7 above)
Comment (Arishamays1) [reply recipe/meta.yaml:33]: Removed pinning. Thanks!

**Review by marcoesters — APPROVED** [review]
LGTM

**Review by varlackc — APPROVED** [review]
LGTM
